pypsa/clustering/spatial.py

```diff
@@ -57,12 +57,22 @@ def consense(x):
 
 
 def aggregategenerators(
-    network, busmap, with_time=True, carriers=None, custom_strategies=dict()
+    network,
+    busmap,
+    with_time=True,
+    carriers=None,
+    custom_strategies=dict(),
+    aggregate_buses=None,
 ):
     if carriers is None:
         carriers = network.generators.carrier.unique()
 
     gens_agg_b = network.generators.carrier.isin(carriers)
+
+    if aggregate_buses is not None:
+        # add all generators in buses that should be clustered
+        gens_agg_b += network.generators.bus.isin(aggregate_buses)
```

Comment:
Shouldn't it be `|=` since it is boolean series?

Reply:
They seem to work the same way, I find `+=` a bit more intuitive, and found myself looking up `|=` once upon a time but I am happy about either.